src/components/Posts.context.js

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, Component } from 'react'
+
+const PostsContext = createContext({
+    posts: [],
+    subreddit: '',
+    isFetching: false,
+    selectedSubreddit: () => {}
+})
+
+export const PostsConsumer = PostsContext.Consumer
+
+export const postsStore = Component => props => (
+    <PostsConsumer>
+        {store => <Component {...props} {...store} />}
+    </PostsConsumer>
+)
+
+class PostsProvider extends Component {
+    state = {
+        posts: [],
+        subreddit: 'reactjs',
+        isFetching: false,
+        selectedSubreddit: (subreddit) => this.fetchPosts(subreddit)
```

Comment:
Copy of the previous comment

```suggestion
        selectSubreddit: (subreddit) => this.fetchPosts(subreddit)
```